simulariumio/custom_converter.py

```diff
@@ -1,22 +1,56 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
 
+import json
 import logging
-from typing import Any, Dict
+from typing import Any, Dict, List
+import math
 
 import numpy as np
 
-from .converter import Converter
-from .data_objects import CustomData
+from .plot_readers import (
+    HistogramPlotReader,
+    ScatterPlotReader,
+    PlotReader,
+)
+from .data_objects import (
+    HistogramPlotData,
+    ScatterPlotData,
+    AgentData,
+    CustomData,
+)
+from .filters import (
+    EveryNthAgentFilter,
+)
+from .filters.params import FilterParams
+from .filters.filter import Filter
+from .exceptions import (
+    UnsupportedPlotTypeError,
+    UnsupportedFilterTypeError,
+)
+from .constants import V1_SPATIAL_BUFFER_STRUCT, VIZ_TYPE
 
 ###############################################################################
 
 log = logging.getLogger(__name__)
 
 ###############################################################################
 
+SUPPORTED_PLOT_READERS = {
+    "scatter": ScatterPlotReader,
+    "histogram": HistogramPlotReader,
+}
+
+FILTERS = {
+    "every_nth_agent": EveryNthAgentFilter,
```

Comment:
can users make their own filters that don't live in this repo?

Reply:
no they can't, but I should fix that